### PR TITLE
feat: curio: web based config edit

### PR DESCRIPTION
**Pull request by snadrus**
## Related Issues
1. guided-setup for migrations sometimes does not bring over the API key
2. WARN if no POST worker is present for the miner IDs we are servicing. 
3. Fix docs generation for curio-cli.
4. Config Editor 
image
Screenshot 2024-04-05 at 10 54 20 PM

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by magik6k — COMMENTED** [review]
not much to be nitpicky about, looks good if it works

File: curiosrc/web/static/ux/curio-ux.mod.js
[the remark this answers is not in the document]
Comment (snadrus) [reply curiosrc/web/static/ux/curio-ux.mod.js:1]: .mod.js opens correctly in VSCode, but .jsm does not.

[the remark this answers is not in the document]
Comment (snadrus) [reply curiosrc/web/static/ux/curio-ux.mod.js:1]: Also jsm sends the wrong mime type in Go which causes the browser to not load the JS.
Mozilla likes mjs which I find quite confusing.

File: curiosrc/web/static/chain-connectivity.jsm
Comment (magik6k) [inline curiosrc/web/static/chain-connectivity.jsm:1]: In that case do we want to undo this move / make it .mod.js?

File: curiosrc/web/static/ux/curio-ux.mod.js
[the remark this answers is not in the document]
Comment (snadrus) [reply curiosrc/web/static/ux/curio-ux.mod.js:1]: lets go with the 3rd option, mjs. It's respected by VSCode and Go HTTP.

File: curiosrc/web/static/chain-connectivity.jsm
Comment (snadrus) [reply curiosrc/web/static/chain-connectivity.jsm:1]: mjs works in VSCode & Go HTTP and communicates the most succinctly.

**Review by LexLuthr — APPROVED** [review]
Ready to 🚢

**Pull request merged**